test/app-luatest/tap_test.lua

```diff
@@ -0,0 +1,633 @@
+-- The test for The Test Anything Protocol module.
+--
+-- Disable stack traces for this test because Tarantool test
+-- system also checks test output.
+
+local t = require('luatest')
+local treegen = require('luatest.treegen')
+local justrun = require('luatest.justrun')
+
+local g = t.group()
+
+-- The testcase tests `ok`, `fail` and `skip` predicates.
+g.test_tap_module_test_statuses = function()
+    local dir = treegen.prepare_directory({}, {})
+    local script_name = 'tap_test_tap_module_test_statuses.lua'
+    treegen.write_file(dir, script_name, [[
+local tap = require('tap')
+local test = tap.test('root test')
+test.trace = false
+test:plan(5)
+test:ok(true, 'true') -- basic function
+local extra = { state = 'some useful information to debug on failure',
+        details = 'a table argument formatted using yaml.encode()' }
+test:ok(true, "extra information is not printed on success", extra)
+test:ok(false, "extra printed using yaml only on failure", extra)
+
+test:fail('failed') -- always fail the test
+test:skip('test marked as ok and skipped')
+    ]])
+    local opts = {nojson = true, stderr = true}
```

Comment:
Nit: seems like the `stderr = true` option is not needed (because it is not checked in the test cases).

Reply:
Thanks!

removed `stderr = true` here and below

```diff
diff --git a/test-run b/test-run
index dc2382c300..d04c5956e7 160000
--- a/test-run
+++ b/test-run
@@ -1 +1 @@
-Subproject commit dc2382c30031471716d8b5dd64e8812441d014aa
+Subproject commit d04c5956e7168a979c84f96f3dd32642eb5d8055
diff --git a/test/app-luatest/tap_test.lua b/test/app-luatest/tap_test.lua
index 05e50d44cb..4bd3623598 100644
--- a/test/app-luatest/tap_test.lua
+++ b/test/app-luatest/tap_test.lua
@@ -27,7 +27,7 @@ test:ok(false, "extra printed using yaml only on failure", extra)
 test:fail('failed') -- always fail the test
 test:skip('test marked as ok and skipped')
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -57,7 +57,7 @@ test:is(tonumber('48'), 48, 'tonumber(48) is 48')
 test:isnt(0xff, 64, '0xff is not 64')
 test:isnt(1, 1, '1 is not 1')
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -101,7 +101,7 @@ test:iscdata(10, 'int', 'cdata type')
 test:iscdata(ffi.new('int', 10), 'int', 'cdata type')
 test:iscdata(ffi.new('unsigned int', 10), 'int', 'cdata type')
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -182,7 +182,7 @@ test:isnt(box.NULL, box.NULL, "box.NULL != box.NULL strict = true")
 test:isnt(nil, nil, "nil != nil strict = true")
 test.strict = false
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -226,7 +226,7 @@ test:test("subtest 1", function(t)
     -- test:check() is called automatically
 end)
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -255,7 +255,7 @@ local sub2 = test:test("subtest 2")
     sub2:diag('hello from subtest')
     sub2:check() -- please call check() explicitly
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -293,7 +293,7 @@ test:test("1 level", function(t)
     end)
 end)
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -336,7 +336,7 @@ test:test("bad plan", function(t)
     t:ok(true, 'true')
 end)
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -366,7 +366,7 @@ test:test("failed subtest", function(t)
     t:fail("failed subtest")
 end)
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -438,7 +438,7 @@ test:test('is_deeply', function(t)
     t.strict = false
 end)
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -554,7 +554,7 @@ test:test('like', function(t)
     t:unlike('abcde', 'acd', 'unlike(abcde, acd)')
 end)
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -582,7 +582,7 @@ test:plan(2)
 test:is_deeply({1, 2, 3}, '200', "compare {1, 2, 3} and '200'")
 test:is_deeply('200', {1, 2, 3}, "compare '200' and {1, 2, 3}")
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
@@ -622,7 +622,7 @@ test:ok(true)
 test:ok(true)
 test:check() -- call check() explicitly.
     ]])
-    local opts = {nojson = true, stderr = true}
+    local opts = {nojson = true}
     local res = justrun.tarantool(dir, {}, { script_name }, opts)
     t.assert_equals(res.exit_code, 0)
     t.assert_equals(res.stdout, [[
```

Reply:
I've seen similar `stderr = true` in other tests. That's minor, so I'll close the comment.

Dropping the `stderr = true` option may report failures better: stderr goes to stderr and captured by luatest for the report.